### ISSUE TITLE
How can I set the consumedir options in a Docker environment?

### ISSUE THREAD

**Issue by papaloewe** [eikek/docspell#714]
I have the problem that every minute the same file is imported again and again.
I thought the .env file was the right way and added the line "CONSUMEDIR_UNIQUE=y ".
However, this seems to be ignored at startup.

**Comment by papaloewe** [comment]
I have found in the Docspell documentation that in a Docker environment the script: "consumedir-cleaner.sh" should be used for this.
This is located in the consumedir container, but is not triggered at all.
Do I see this correctly?
How can I start this correctly and periodically within the consumedir container?

**Comment by eikek** [comment]
Hi @papaloewe , I have to check these scripts a little more; from the top of my head: this should work as you suggested in your first comment. This consumedir script should not upload files again, if they have been uploaded already. This works for me currently, but I'm not using the docker setup myself. If that's not working, then I'd guess right now that the option is not picked up. Do you have any other changes in the `.env` file?

The `consumedir-cleaner.sh` script is for the cases where you have a bunch of files and you want each to be checked whether it is in docspell and if so do something with it: moving away or deleting it.

**Comment by papaloewe** [comment]
This is my .env file:
`TZ=Europe/Berlin
DOCSPELL_HEADER_VALUE="my-secret-password"
DB_TYPE=postgresql
DB_HOST=db
DB_PORT=5432
DB_NAME=dbname
DB_USER=dbuser
DB_PASS=dbpass
CONSUMEDIR_VERBOSE=y
CONSUMEDIR_INTEGRATION=y
CONSUMEDIR_POLLING=y
CONSUMEDIR_DELETE=n
CONSUMEDIR_UNIQUE=y

`

**Comment by papaloewe** [comment]
I think you are right. The option is noch picked, right?
Log from consumedir-container:
`
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret-password" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret-password" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode`

**Comment by eikek** [comment]
Yes, that seems so. I think it might be best to create all the switches yourself. The script only uses the `y`/`n` switches if the env variable `CONSUMEDIR_ARGS` is empty. So we could try to set this variable only and remove all the others. IMHO this is easier.

If you want polling instead of inotify, then only the `-m` or `--distinct` option is missing, I'd say.

**Comment by eikek** [comment]
Update: when using polling, then the `CONSUMEDIR_POLLING=y` is still necessary.

**Comment by eikek** [comment]
More concretely :)
```
CONSUMEDIR_POLLING=y
CONSUMEDIR_ARGS="--path /opt/docs --iheader Docspell-Integration:"my-secret-password" -i -v --once -m http://docspell-restserver:7880/api/v1/open/integration/item"
```
```

**Comment by papaloewe** [comment]
Unfortunately, that doesn't work either.
I wrote both lines in the .env fileil as well as directly in the docker-compose.yml.
However, this is still ignored when starting the comsumdir container.

**Comment by eikek** [comment]
Strange. The file gets uploaded? Can you maybe attach some logs from the consumedir container?

**Comment by papaloewe** [comment]
Yes, the file get uploaded every minute again and again.

Her is some more from the consumedir-container log.
`
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Using default CONSUMEDIR_ENDPOINT=http://docspell-restserver:7880/api/v1/open/integration/item
Command: /opt/docspell-tools/consumedir/consumedir.sh --path /opt/docs --iheader Docspell-Integration:"my-secret" -i -v --once http://docspell-restserver:7880/api/v1/open/integration/item
Running in polling mode
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.`

**Comment by papaloewe** [comment]
And this is the moment when a file is uploaded:
`
Checking integration endpoint: curl  -i -s -o /dev/null -w %{http_code} -H Docspell-Integration:"my-secret" http://docspell-restserver:7880/api/v1/open/integration/item/tom
Uploading all files (except hidden) in '/opt/docs'.
---- Processing /opt/docs/tom/20210215_000327.pdf ----------
- Uploading '/opt/docs/tom/20210215_000327.pdf' to 'http://docspell-restserver:7880/api/v1/open/integration/item'.
- upload: collective = tom
- Uploading /opt/docs/tom/20210215_000327.pdf to http://docspell-restserver:7880/api/v1/open/integration/item/tom with options  -H Docspell-Integration:"my-secret"
- Upload done.
= File processed for all URLs`

**Comment by eikek** [comment]
Hm, still have no idea currently. I just tried with this `.env` file:

```
TZ=Europe/Berlin
DOCSPELL_HEADER_VALUE=123
DB_TYPE=postgresql
DB_HOST=db
DB_PORT=5432
DB_NAME=dbname
DB_USER=dbuser
DB_PASS=dbpass
CONSUMEDIR_VERBOSE=y
CONSUMEDIR_INTEGRATION=y
CONSUMEDIR_POLLING=y
CONSUMEDIR_POLLING_INTERVAL=10
CONSUMEDIR_ARGS="--path /opt/docs --iheader 'Docspell-Integration:123' -i -d -v --once -m http://docspell-restserver:7880/api/v1/open/integration/item"
```

The `-m` gets picked up from what i see in the logs.

**Comment by papaloewe** [comment]
Oh no, my fault. I am sorry about that.
Every try I was typing "docker-compose restart"
Now I made "docker-compose stop" and then "docker compose up-d"
That makes the difference. I did not knew that.

The comsumedir-container started now with that:
`Command: /opt/docspell-tools/consumedir/consumedir.sh "--path /opt/docs --iheader 'Docspell-Integration:my-secret' -i -d -v --once -m http://docspell-restserver:7880/api/v1/open/integration/item"`

Thank you for your help.
